sda-download/api/sda/sda.go

```diff
@@ -29,6 +33,43 @@ func sanitizeString(str string) string {
 	return pattern.ReplaceAllString(str, "[identifier]: $1")
 }
 
+func reencryptHeader(oldHeader []byte, reencKey string) ([]byte, error) {
+	if config.Config.Reencrypt.Host == "" {
+		return nil, fmt.Errorf("host of the reencrypt service not set")
+	} else if config.Config.Reencrypt.Port == 0 {
+		return nil, fmt.Errorf("port of the reencrypt service not set")
+	}
+	address := fmt.Sprintf("%s:%d", config.Config.Reencrypt.Host, config.Config.Reencrypt.Port)
+	log.Debugf("Address of the reencrypt service: %s", address)
+
+	opts := []grpc.DialOption{
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+	}
+	conn, err := grpc.Dial(address, opts...)
+	if err != nil {
+		log.Errorf("Failed to connect to the reencrypt service, reason: %s", err)
+
+		return nil, err
+	}
+	defer conn.Close()
+	log.Debugf("Connection to the reencrypt service established, conn = %v", conn)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
```

Comment:
```suggestion
	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
```

Connection timeout of 1 second is a bit harsh.